components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -0,0 +1,157 @@
+    private static final QName HANDLER_Q = new QName("api");
+    private static final QName CLASS_Q = new QName("class");
+    private static final QName NAME_ATT = new QName("name");
+
+    public static List<InternalAPI> loadInternalAPIs(String apiFilePath) {
+
+        List<InternalAPI> internalApis = new ArrayList<>();
+        OMElement apiConfig = MiscellaneousUtil.loadXMLConfig(apiFilePath);
+        if (apiConfig != null) {
+
+            if (!ROOT_Q.equals(apiConfig.getQName())) {
+                handleException("Invalid internal api configuration file");
+            }
+
+            Iterator iterator = apiConfig.getChildrenWithName(HANDLER_Q);
+            while (iterator.hasNext()) {
+                OMElement handlerElem = (OMElement) iterator.next();
+
+                String name = null;
+                if (handlerElem.getAttribute(NAME_ATT) != null) {
+                    name = handlerElem.getAttributeValue(NAME_ATT);
+                } else {
+                    handleException("Name not defined in one or more handlers");
+                }
+
+                if (handlerElem.getAttribute(CLASS_Q) != null) {
+                    String className = handlerElem.getAttributeValue(CLASS_Q);
+                    if (!className.isEmpty()) {
+                        InternalAPI internalApi = createAPI(className);
+                        if (internalApi != null) {
+                            internalApis.add(internalApi);
+                            internalApi.setName(name);
+                        }
+                    } else {
+                        handleException("Class name is null for Internal InternalAPI name : " + name);
+                    }
+                } else {
+                    handleException("Class name not defined for Internal InternalAPI named : " + name);
+                }
+            }
+        }
+        return internalApis;
+    }
+
+    private static InternalAPI createAPI(String classFQName) {
+
+        Object obj = null;
+        try {
+            obj = Class.forName(classFQName).newInstance();
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            handleException("Error creating Internal InternalAPI for class name : " + classFQName, e);
+        }
+
+        if (obj instanceof InternalAPI) {
+            return (InternalAPI) obj;
+        } else {
+            handleException("Error creating Internal InternalAPI. The InternalAPI should be of type " +
+                    "InternalAPI");
+        }
+        return null;
+    }
+
+
+    public static int getInternalInboundPort() {
+
+        File synapseProperties = Paths.get(CarbonUtils.getCarbonConfigDirPath(), "synapse.properties").toFile();
+        Properties properties = new Properties();
+        InputStream inputStream;
+        try {
+            inputStream = new FileInputStream(synapseProperties);
+            properties.load(inputStream);
+            inputStream.close();
```

Comment:
shouldn't we close this in finally block or instantiate in try block itself

Reply:
yeah, did the change.